src/classes/service-proxy/service-proxy.cpp

```diff
@@ -0,0 +1,73 @@
+#include "service-proxy.hpp"
+
+auto Status::getState() -> Status::stateT
+{
+
+    const std::lock_guard<std::mutex> lock(m_mtx);
+
+    return m_state;
+}
+
+void Status::setState(Status::stateT newState)
+{
+
+    const std::lock_guard<std::mutex> lock(m_mtx);
+
+    m_state = newState;
+}
+
+void ServiceProxy::servicePod(IService& service, Status& status)
+{
+
+    service.setup();
+    status.setState(Status::RUNNING);
+
+    while (status.getState() != Status::STOPED) {
```

Comment:
Aqui não seria melhor se fosse ?
```suggestion
    while (status.getState() == Status::RUNNING) 
```
Existem outros estados além desses e o único que permite que ela esteja de fato rodando é o running, não?

Reply:
A adoção do `!= Status::STOPPED` tem o objetivo de reservar ao status STOPPED - e somente a ele - o direito de encerrar "rotinas". Assim, a idea era que cada "state" tivesse um objetivo no controle da thread, permitindo ao programador por meio da requisição de status, saber exatamente o que "está acontecendo"/"aconteceu" com cada um de seus "services".